client/js/app.js

```diff
@@ -10,24 +10,65 @@ var Day = require('./components/day/day.js');
 var Week = require('./components/week/week.js');
 var Actionbar = require('./components/actionbar/action-bar.js');
 var Sidebar = require('./components/sidebar/sidebar.js');
+var classnames = require('classnames');
 
 var Locations = Router.Locations;
 var Location = Router.Location;
 var NotFound = Router.NotFound;
 
 var App = React.createClass({
+  getInitialState: function () {
+    return {
+      actionbarOpen: false,
+      hashRoute: ''
+    };
+  },
+
+  componentDidMount: function () {
+    window.addEventListener('hashchange', this.handleHashChange);
+  },
+
+  handleHashChange: function () {
+    var newHash = window.location.hash;
+    var actionbarHashActivated = /#actionbar?.+/.test(newHash);
+    var actionbarOpen = this.state.actionbarOpen;
+
+    if (!actionbarOpen && !actionbarHashActivated) {
```

Comment:
Yea good call. I was overprotecting for all route changes but hash is the only thing that matters here.